### PR TITLE
Adds `uv` step

### PR DESCRIPTION
**Pull request by lucaspar**
This `uv` step runs global update tasks for the relatively new [`uv`](https://github.com/astral-sh/uv) python package and project manager.

On update, two actions are executed:

1. `uv self update` [docs](https://docs.astral.sh/uv/getting-started/installation/#standalone-installer)
2. `uv tool upgrade --all` [docs](https://docs.astral.sh/uv/guides/tools/#installing-tools)

## Standards checklist:

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] The code compiles (`cargo build`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [x] *Optional:* I have tested the code myself
    - `cargo run -- --verbose --only uv` works.
 
## For new steps

- [x] *Optional:* Topgrade skips this step where needed
- [x] *Optional:* The `--dry-run` option works with this step
- [ ] *Optional:* The `--yes` option works with this step if it is supported by 
  the underlying command
    - underlying command does not support `--yes`.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -327,6 +327,7 @@ fn run() -> Result<()> {
             unix::upgrade_gnome_extensions(&ctx)
         })?;
         runner.execute(Step::Pyenv, "pyenv", || unix::run_pyenv(&ctx))?;
+        runner.execute(Step::Uv, "uv", || unix::run_uv(&ctx))?;
```
Comment (SteveLauC) [inline src/main.rs:330]: Looks like this tool is also available on Windows, we should probably move the impl to `general.rs`👀

**Review by SteveLauC — APPROVED** [review]
LGTM, just one tiny issue. 😁

**Comment by lucaspar** [comment]
@SteveLauC I think that should do it.

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**